asyncapi-cli/src/main/java/io/ballerina/asyncapi/cmd/AsyncApiCmd.java

```diff
@@ -105,11 +149,41 @@ public void execute() {
                 return;
             }
             String fileName = argList.get(0);
-            Application codeGenerator = new CodeGenerator();
-            try {
-                codeGenerator.generate(fileName, (outputPath == null) ? String.valueOf(executionPath) : outputPath);
-            } catch (BallerinaAsyncApiException e) {
-                outStream.println(e.getMessage());
+
+            if (protocol.equals("http")) {
+                verifyValidInputsForHttp();
+                Application codeGenerator = new CodeGenerator();
+                try {
+                    codeGenerator.generate(fileName, (outputPath == null) ?
+                            String.valueOf(executionPath) : outputPath);
+                } catch (BallerinaAsyncApiException e) {
+                    outStream.println(e.getMessage());
+                    exitError(this.exitWhenFinish);
+                }
+            } else if (protocol.equals("ws") || protocol.equals("wss") || protocol.equals("websocket")) {
+                if (fileName.endsWith(Constants.YAML_EXTENSION) || fileName.endsWith(Constants.JSON_EXTENSION) ||
+                        fileName.endsWith(Constants.YML_EXTENSION)) {
+                    try {
+                        asyncApiToBallerinaWs(fileName);
+                    } catch (IOException e) {
+                        outStream.println(e.getLocalizedMessage());
+                        exitError(this.exitWhenFinish);
```

Comment:
Rather than doing a runtime exit, shall we create an error diagnostic and return?